src/Compilers/Core/MSBuildTaskTests/RuntimeHostInfoTests.cs

```diff
@@ -13,75 +14,99 @@
 
 namespace Microsoft.CodeAnalysis.BuildTasks.UnitTests;
 
-public sealed class RuntimeHostInfoTests(ITestOutputHelper output)
+public sealed class RuntimeHostInfoTests(ITestOutputHelper output) : TestBase
 {
     private readonly ITestOutputHelper _output = output;
 
-    [Fact, WorkItem("https://github.com/dotnet/msbuild/issues/12669")]
-    public void DotNetInPath()
+    /// <summary>
+    /// Normalizes a path by resolving any symbolic links or subst drives (on Windows).
+    /// This ensures paths can be compared even when one is a subst drive (e.g., T:\ -> D:\a\_work\1\s\artifacts\tmp\Debug).
+    /// </summary>
+    private string NormalizePath(string path)
     {
-        var previousPath = Environment.GetEnvironmentVariable("PATH");
         try
         {
-            using var tempRoot = new TempRoot();
-            var testDir = tempRoot.CreateDirectory();
-            var globalDotNetDir = testDir.CreateDirectory("global-dotnet");
-            var globalDotNetExe = globalDotNetDir.CreateFile($"dotnet{PlatformInformation.ExeExtension}");
-            Environment.SetEnvironmentVariable("PATH", globalDotNetDir.Path);
-
-            Assert.Equal(globalDotNetDir.Path, RuntimeHostInfo.GetToolDotNetRoot(_output.WriteLine));
+            // Our test infra uses `subst` which the .NET Core's implementation of `ResolveLinkTarget` wouldn't resolve,
+            // hence we always use our Win32 polyfill on Windows to ensure paths are fully normalized and can be compared in tests.
+            var resolvedPath = PlatformInformation.IsWindows
+                ? NativeMethods.ResolveLinkTargetWin32(path, returnFinalTarget: true)
+                : File.ResolveLinkTarget(path, returnFinalTarget: true);
+            if (resolvedPath != null)
+            {
+                return resolvedPath.FullName;
+            }
         }
-        finally
+        catch (Exception ex)
         {
-            Environment.SetEnvironmentVariable("PATH", previousPath);
+            // If resolution fails, use the original path
```

Comment:
if things fail... should we just fail entirely?  unclear why we're falling back in that case.